pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -760,73 +760,28 @@ protected SegmentBuildDescriptor buildSegmentInternal(boolean forCommit) {
     }
   }
 
-  private SegmentCompletionProtocol.Response doSplitCommit(SegmentCompletionProtocol.Response prevResponse) {
-    final File segmentTarFile = new File(_segmentBuildDescriptor.getSegmentTarFilePath());
+  protected boolean commitSegment(SegmentCompletionProtocol.Response response) {
```

Comment:
Extract value of `isSplitCommit` and `controllerVipUrl` from response, and pass those as arguments into the `commitSegment` method.